modules/repository-azure/build.gradle

```diff
@@ -17,32 +17,47 @@ esplugin {
 }
 
 versions << [
-  'azure': '12.20.1',
-  'azureCommon': '12.19.1',
-  'azureCore': '1.34.0',
-  'azureCoreHttpNetty': '1.12.7',
-  'azureJackson': '2.13.4',
-  'azureJacksonDatabind': '2.13.4.2',
-  'azureAvro': '12.5.3',
-
-  'jakartaActivation': '1.2.1',
-  'jakartaXMLBind': '2.3.2',
+  //see https://repo1.maven.org/maven2/com/azure/azure-storage-blob/ for transitive dependencies
+  'azure': '12.26.1', // <-- when upgrading versions start start here and manually upgrade the transitive dependencies
+  'azureCommon': '12.25.1',
+  'azureCore': '1.49.1',
+  'azureCoreHttpNetty': '1.15.1',
+  'azureJackson': '2.15.4', // keep this roughly in sync with the jackson version in the rest of the codebase
+  'azureAvro': '12.11.1',
+
+  // see https://repo1.maven.org/maven2/com/azure/azure-core/
+  'azureJson': '1.1.0',
+  'azureXml': '1.0.0',
```

Comment:
azure-xml and azure-json are 2 new dependencies brought in via azure-core. see release notes here: https://repo1.maven.org/maven2/com/azure/azure-core/1.49.1/azure-core-1.49.1-changelog.md 

> ## 1.48.0 (2024-04-05)
> ...
> - Added dependency on `azure-xml` to support `XmlSerializable`, removing the need for Jackson Databind XML to handle
> - Made `azure-json` `requires transitive` in `module-info.java`.
>...

I am pretty confident that the Jackson jars are still needed since it is still explicitly listed as allowed : https://repo1.maven.org/maven2/com/azure/azure-core/1.49.1/azure-core-1.49.1.pom (note "[included](https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html)" in maven config is IMO misleading), and jackson-dataformat-xml is a compile time dependency for [azure-storage-blob](https://repo1.maven.org/maven2/com/azure/azure-storage-blob/12.26.1/azure-storage-blob-12.26.1.pom)

Reply:
I also found this: https://learn.microsoft.com/en-us/java/api/overview/azure/xml-readme?view=azure-java-preview and https://github.com/Azure/azure-sdk-for-java/pull/39563 , which leads me to believe they are slowing removing the need to use Jackson.